### PR TITLE
[WIP] Update core plugin example

### LINKED ISSUE #1134: Update currently broken core_plugin_example.c to Newshell API
The example core plugin at https://github.com/rizinorg/rizin/blob/bdface85a24a52d1c4a8cf2daa00dc6099e5db51/librz/core/p/core_plugin_example.c still uses a `.call = rz_cmd_example_call` callback for injecting commands, which does not exist anymore.

For reference, here are some plugins that have been updated:
https://git.sr.ht/~thestr4ng3r/rz-commodore/tree/master/item/src/core_commodore.c
https://github.com/rizinorg/rz-ghidra/blob/dev/src/core_ghidra.cpp
https://github.com/rizinorg/jsdec/blob/master/p/core_pdd.c

It might also be nice to somehow embed this in our build system and add to the test suite in some way.

### PR DESCRIPTION
**Pull request by pparmin**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

I have updated the core plugin example according to the Newshell API. I wasn't entirely sure whether I was supposed to also add an example for how to register an exemplary command, but I went ahead and did it anyways. Please let me know if it should be removed :) 

I also think that this might need more/new comments so I wanted to see what everyone's idea was on that. 

...

**Test plan**
In the original issue it was mentioned that this could maybe also be embedded in the build system and the test suite. What could be possible next steps for that? 

...
closes #1134 
**Closing issues**

...

### CODE REVIEW

File: librz/core/p/core_plugin_example.c [hunk 1]
```diff
@@ -18,23 +18,41 @@ mv core_plugin_example.so ~/.config/rizin/plugins
...
+static const RzCmdDescHelp example_usage = {
+	.summary = "Example core plugin",
+};
+
+RZ_IPI RzCmdStatus rz_cmd_example_handler(RzCore *core, int argc,
+	const char **argv) {
+	if (argc < 2) {
+		return RZ_CMD_STATUS_WRONG_ARGS;
+	}
+	eprintf("This example was called!\n")
```
Comment (officialcjunior) [inline librz/core/p/core_plugin_example.c]: I think you forgot to put a semicolon?

```suggestion
	eprintf("This example was called!\n");
```

**Review by ret2libc — COMMENTED** [review]
@thestr4ng3r  I was wondering.... do we really need this file? Maybe we should just create a unit test for core plugins in test/unit? After all, unit tests are supposed to be used also as examples, aren't they?

File: librz/core/p/core_plugin_example.c (hunk 1 above)
Comment (pparmin) [reply librz/core/p/core_plugin_example.c]: classic mistake on the first PR :D I will fix it!

**Comment by XVilka** [comment]
@pparmin thank you for your contribution, but this PR addresses a few more issues with the plugin example: https://github.com/rizinorg/rizin/pull/1235/files#diff-9c5355c5305f7490a4f08f047c72728f24cdd71333af792ce33f55b5c65d0e3b

**Comment by XVilka** [comment]
Fixed in https://github.com/rizinorg/rizin/commit/8fe4d0b0a54ec1fa6dfc8bcc497bacb6ee5f00ee

**Pull request closed without merging**